contracts/contracts/gateway/router/CheckpointingFacet.sol

```diff
@@ -58,66 +40,14 @@ contract CheckpointingFacet is GatewayActorModifiers {
         _execBottomUpMsgBatch(msgs, subnet);
     }
 
-    /// @notice creates a new bottom-up checkpoint
-    /// @param checkpoint - a bottom-up checkpoint
-    /// @param membershipRootHash - a root hash of the Merkle tree built from the validator public keys and their weight
-    /// @param membershipWeight - the total weight of the membership
-    /// @param msgs - the full messages batch
-    /// @param activity - the full activity rollup
-    function createBottomUpCheckpoint(
-        BottomUpCheckpoint calldata checkpoint,
-        bytes32 membershipRootHash,
-        uint256 membershipWeight,
+    function recordLightClientCommitments(
+        StateCommitmentBreakDown calldata commitment,
```

Comment:
I'm not sure why it makes sense to emit 3 events here. 
Previously storage was used to maintain the checkpoint while signatures were being added.
Now this process is eliminated, so similarly to how you aggregated all commitments to one struct, the same can be done here with these events. 

In addition, I don't think it makes sense to record the checkpoint with missing data: the signed header. Requiring the relayer to fetch it from cometbft is problematic, because this functionality should be encapsulated. Similarly, we can avoid these events altogether, requiring the relayer to scrape all needed information -- but I do think it make sense to mark the checkpoint with an event, only that it should contain all info for any party to pick it and submit it as-is.

Reply:
If relayer does not query from cometbft, I'm not sure where this information can be retrieved. 

As for the 3 events, are you suggesting emit as one? Or dont emit at all?